extensions/amp-story-interactive/0.1/amp-story-interactive-abstract.js

```diff
@@ -857,4 +869,78 @@ export class AmpStoryInteractive extends AMP.BaseElement {
 
     return orderedData;
   }
+
+  /**
+   * Opens the disclaimer dialog and positions it according to the page and itself.
+   */
+  openDisclaimer_() {
+    if (this.disclaimerEl_) {
+      return;
+    }
+    this.disclaimerEl_ = buildInteractiveDisclaimer(this);
+
+    const pageEl = closest(
+      dev().assertElement(this.element),
+      (el) => el.tagName.toLowerCase() === 'amp-story-page'
+    );
+
+    let styles;
+    this.measureMutateElement(
+      () => {
+        // Get rects and calculate position from icon.
+        const interactiveRect = this.element./*OK*/ getBoundingClientRect();
+        const pageRect = pageEl./*OK*/ getBoundingClientRect();
+        const iconRect = this.disclaimerIcon_./*OK*/ getBoundingClientRect();
+        const rightFraction =
+          1 - (iconRect.x + iconRect.width - pageRect.x) / pageRect.width;
+        const bottomFraction =
+          1 - (iconRect.y + iconRect.height - pageRect.y) / pageRect.height;
+        const widthFraction = interactiveRect.width / pageRect.width;
+
+        // Clamp values to ensure dialog has space up and left.
+        const rightPercentage = clamp(rightFraction * 100, 0, 25); // Ensure 75% of space to the left.
+        const bottomPercentage = clamp(bottomFraction * 100, 0, 85); // Ensure 15% of space up. 
+        const widthPercentage = Math.max(widthFraction * 100, 65); // Ensure 65% of max-width.
+
+        styles = {
+          'bottom': bottomPercentage + '%',
+          'right': rightPercentage + '%',
+          'max-width': widthPercentage + '%',
+          'position': 'absolute',
+          'z-index': 3,
```

Comment:
I wonder if these could be part of `i-amphtml-story-interactive-disclaimer-dialog`.

Reply:
I can add them to the container CSS (which would have to be defined on `amp-story.css`) but not the dialog because the shadow root is in the way, so the positioning is done on the container of the shadow root. The reason why I didn't move these 2 styles to the CSS is because they would be extra payload on all stories, while the styles here are only loaded with the interactive components and are kept all in one place